vonk/plugins/plugins_order.rst

```diff
@@ -0,0 +1,86 @@
+.. _vonk_plugins_order:
+
+The order of plugins
+====================
+
+The :ref:`VonkConfiguration attribute <vonk_reference_api_pipeline_configuration>` lets you define an ``Order``. This page explains how to choose a suitable number for that order.
+
+.. _vonk_plugins_order_inspect:
+
+Inspect order numbers in use
+----------------------------
+
+If you start Vonk, the log lists all the loaded plugins, with their order. You can see an example :ref:`here <vonk_plugins_log_pipeline>`. Also the list of :ref:`vonk_available_plugins` includes the order number chosen for each of those plugins.
+
+.. _vonk_plugins_order_minimum:
+
+Minimum order
+-------------
+
+Registering services
+^^^^^^^^^^^^^^^^^^^^
+
+If you only need to register services for the dependency injection framework, the order is only relevant if you need to override a registration done by Vonk. There are two ways:
+
+1. Choose an order before Vonk's default registration. Vonk in general uses ``TryAddSingleton`` or ``TryAddScoped`` to register an implementation of an interface. This means that if an implementation is already registered, the TryAdd... will not register a second implementation.
+
+   As an example: If you want to override the registration of ``IReadAuthorizer``: that is registered from the :ref:`RepositorySearchSupport <vonk_plugins_search>` plugin, with order 140. So you would choose an order lower than 140.
+
+2. Choose a high order (e.g. > 10000) and make sure your registration overwrites any existing registration.
+
+   ``services.AddOrReplace<IReadAuthorizer, MyReadAuthorizer>(ServiceLifetime.Scoped);``
+
+If you only need to register interfaces and/or classes defined by your plugin, the order is not relevant. All service registrations are done before the pipeline itself is configured.
+
+Registering middleware
+^^^^^^^^^^^^^^^^^^^^^^
+
+For middleware it is more important where exactly it ends up in the pipeline. This depends mostly on what type of handler it is, see below at :ref:`vonk_plugins_order_prepost`. 
+
+No matter what handler you have, it probably wants to act on the IVonkContext. Then it is important to be in the pipeline *after* the :ref:`HttpToVonk <vonk_plugins_httptovonk>` plugin (order: 1110), since this plugin translates information from the ``HttpContext`` to an ``IVonkContext`` and adds the latter as a feature to the ``HttpContext.Features`` collection. 
```

Comment:
```suggestion
No matter what handler you have, it probably wants to act on the IVonkContext: then it is important to be in the pipeline *after* the :ref:`HttpToVonk <vonk_plugins_httptovonk>` plugin (order: 1110), since this plugin translates information from the ``HttpContext`` to an ``IVonkContext`` and adds the latter as a feature to the ``HttpContext.Features`` collection. 
```